streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -53,6 +54,7 @@
     private final AssignedStandbyTasks standby;
     private final ChangelogReader changelogReader;
     private final String logPrefix;
+    private final Producer<byte[], byte[]> threadProducer;
```

Comment:
We only initialize threadProducer when the EOS is turned on?

Reply:
In this PR (that will outdate when we merge the refactoring), we only pass in the `threadProducer` for EOS-beta -- for eos-alpha or non-eos it would be `null`